finetuner/hubble.py

```diff
@@ -31,44 +49,62 @@ def push_data(
     :param index_data: Index data for `EvaluationCallback`.
     :return: Name(s) of the uploaded data.
     """
-
-    def _push_docarray(
-        data: Union[None, str, DocumentArray], name: str, ids2names: Dict[int, str]
-    ) -> Optional[str]:
-        if isinstance(data, DocumentArray):
-            _id = id(data)  # get the reference id
-            if _id in ids2names:
-                return ids2names[_id]
-            print(f'Pushing a DocumentArray to Hubble under the name {name} ...')
-            data.push(name=name, show_progress=True, public=False)
-            ids2names[id(data)] = name
-            return name
-        return data
-
     _ids2names = dict()
     return (
-        _push_docarray(
+        push_docarray(
             train_data, f'{DA_PREFIX}-{experiment_name}-{run_name}-train', _ids2names
         ),
-        _push_docarray(
+        push_docarray(
             eval_data, f'{DA_PREFIX}-{experiment_name}-{run_name}-eval', _ids2names
         ),
-        _push_docarray(
+        push_docarray(
             query_data, f'{DA_PREFIX}-{experiment_name}-{run_name}-query', _ids2names
         ),
-        _push_docarray(
+        push_docarray(
             index_data, f'{DA_PREFIX}-{experiment_name}-{run_name}-index', _ids2names
         ),
     )
 
 
+def push_synthesis_data(
+    experiment_name: str,
+    run_name: str,
+    query_data: Union[str, DocumentArray],
+    corpus_data: Union[str, DocumentArray],
+) -> Tuple[Optional[str], ...]:
```

Comment:
maybe better explicit because they are only two
```suggestion
) -> Tuple[Optional[str], Optional[str]]:
```

Reply:
Is it actually possible that it some of them are `None`?